dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,319 @@
+        """
+        return self.get_current_block() - last_sent_block > BLOCK_THRESHOLD
+
+    def get_network_string(self) -> str:
+        """Get the network string for this blockchain. This is what's included in l5 blocks or sent to matchmaking
+        Returns:
+            Network string
+        """
+        return f"binance {'testnet' if self.testnet else 'mainnet'}"
+
+    def get_private_key(self) -> str:
+        """Get the private key for this network
+        Returns:
+            string of the private key
+        """
+        return self.priv_key
+
+    def create_signed_transaction(self, raw_txn: Dict[str, Any]) -> str:
+        """Sign a transaction for this network
+        Args:
+            raw_transaction: The dictionary of the raw transaction containing:
+                amt: amount of BNB to be sent
+                to_address: address to send BNB to
+                memo: string of arbitrary data to add with this transaction
+        Returns:
+            Hex string of the signed transaction
+        """
+
+        transfer_msg = TransferMsg(
+            wallet=self.wallet,
+            symbol="BNB",
+            amount=raw_txn["amount"],
+            to_address=raw_txn["to_address"],
+            memo=raw_txn["memo"],
+            # STOP LINTING!
+        )
+
+        try:
+            _log.info(f"[BINANCE] Signing raw transaction: {transfer_msg}")
+            signed_transaction = Signature(transfer_msg).sign()
+            return signed_transaction
+        except Exception as e:
+            raise exceptions.BadRequest(f"Error signing transaction: {e}")
+
+    def _publish_transaction(self, transaction_payload: str) -> str:
+        """Publish a transaction to this network with a certain data payload
+        Args:
+            transaction_payload: The arbitrary data to send with this transaction
+        Returns:
+            The string of the published transaction hash
+        """
+        _log.info(f"[BINANCE] Publishing transaction. payload = {transaction_payload}")
+        # create and sign transaction data
+        signed_transaction = self.create_signed_transaction(literal_eval(transaction_payload))
+        # Send signed transaction
+        response = self._call_node_rpc("broadcast_tx_commit", {"tx": signed_transaction})
+        return response["result"]["hash"]  # transaction hash
+
+    def _call_node_api(self, method: str, params: Dict[str, Any]) -> Any:
+        full_address = self.api_port + method
+        r = requests.get(full_address, params, timeout=30)
+        error_status = f"Error from binance node with http status code {r.status_code} | {r.text}"
+        if r.status_code != 200:
+            raise exceptions.APIError(error_status)
+        response = r.json()
+        error_response = "The API server call got an error response: {response}"
+        if response.get("error") or response.get("errors"):
+            raise exceptions.APIError(error_response)
+        return response
```

Comment:
use another function so you don't duplicate this code

Reply:
Updated / fixed.